route-rs-runtime/src/composite/mton_composite.rs

```diff
@@ -0,0 +1,157 @@
+        );
+        assert_ne!(queue_capacity, 0, "queue capacity must be non-zero");
+
+        MtoNCompositeBuilder {
+            in_streams: self.in_streams,
+            join_queue_capacity: queue_capacity,
+            tee_queue_capacity: self.tee_queue_capacity,
+            branches: self.branches,
+        }        
+    }
+
+    /// Changes tee_queue_capcity, default value is 10.
+    /// Valid range is 1..=1000
+    pub fn tee_queue_capacity(self, queue_capacity: usize) -> Self {
+        assert!(
+            queue_capacity <= 1000,
+            format!("queue_capacity: {} > 1000", queue_capacity)
+        );
+        assert_ne!(queue_capacity, 0, "queue capacity must be non-zero");
+
+        MtoNCompositeBuilder {
+            in_streams: self.in_streams,
+            join_queue_capacity: self.join_queue_capacity,
+            tee_queue_capacity: queue_capacity,
+            branches: self.branches,
+        }             
+    }
+
+    pub fn branches(self, branches: usize) -> Self {
+        assert!(
+            branches <= 1000,
+            format!("compsite branches: {} > 1000", branches)
+        );
+        assert_ne!(branches, 0, "branches must be non-zero");
+
+        MtoNCompositeBuilder {
+            in_streams: self.in_streams,
+            join_queue_capacity: self.join_queue_capacity,
+            tee_queue_capacity: self.tee_queue_capacity,
+            branches: Some(branches),
+        }
+    }    
+}
+
+impl<Packet: Sized + Send + Clone + 'static> LinkBuilder<Packet, Packet> for MtoNCompositeBuilder<Packet> {
+    fn ingressors(self, in_streams: Vec<PacketStream<Packet>>) -> Self {
+        assert!(
+            in_streams.len() > 1 && in_streams.len() <= 1000,
+            format!("Input streams {} not in 2..=1000", in_streams.len())
+        );
+        MtoNCompositeBuilder {
+            in_streams: Some(in_streams),
+            join_queue_capacity: self.join_queue_capacity,
+            tee_queue_capacity: self.tee_queue_capacity,
+            branches: self.branches,
+        }
+    }
+
+    fn build_link(self) -> Link<Packet> {
+        if self.in_streams.is_none() {
+            panic!("Cannot build link! Missing input stream");
+        } else if self.branches.is_none() {
+            panic!("Cannot build link! Missing number of branches");            
+        } else {
+            let (join_runnables, join_egressors) = JoinLinkBuilder::new()
+                .ingressors(self.in_streams.unwrap())
+                .queue_capacity(self.join_queue_capacity)
+                .build_link();
+            let (mut tee_runnables, tee_egresssors) = TeeLinkBuilder::new()
+                .ingressors(join_egressors)
+                .queue_capacity(self.tee_queue_capacity)
+                .branches(self.branches.unwrap())
+                .build_link();
+            tee_runnables.extend(join_runnables);
+            (tee_runnables, tee_egresssors)
```

Comment:
nit: Could we use `Vec::append` rather than `extend` to make it clear that we're returning all the runnables? I'd even like an extra name like:

```
let all_runnables = tee_runnables.append(join_runnables);
(all_runnables, tee_egressors)
```

Reply:
I suppose, I don't object to append.

Reply:
done